plugins/outputs/influxdb_v2/http.go

```diff
@@ -319,10 +321,11 @@ func (c *httpClient) getRetryDuration(headers http.Header) time.Duration {
 			// there was a value but we couldn't parse it? guess minimum 10 sec
 			retryAfterHeader = 10
 		}
+		// protect against excessively large retry-after
+		retryAfterHeader = math.Max(retryAfterHeader, defaultMaxWaitRetryAfterSeconds)
 	}
-	// take the highest value from both, but not over the max wait.
+	// take the highest value of backoff and retry-after.
 	retry := math.Max(backoff, retryAfterHeader)
-	retry = math.Min(retry, defaultMaxWait)
 	return time.Duration(retry) * time.Second
```

Comment:
Because of the rounding that happens here when converting to time.Duration (int64), the first 6 retries have a retryDuration of `0s` meaning there's no backoff at all.